connectors/uplynk/src/main/java/com/theoplayer/android/connector/uplynk/internal/UplynkEventDispatcher.kt

```diff
@@ -0,0 +1,24 @@
+package com.theoplayer.android.connector.uplynk.internal
+
+import android.os.Handler
+import android.os.Looper
+import com.theoplayer.android.connector.uplynk.UplynkListener
+import com.theoplayer.android.connector.uplynk.network.AssetInfoResponse
+import com.theoplayer.android.connector.uplynk.network.PreplayResponse
+import java.util.concurrent.CopyOnWriteArrayList
+
+internal class UplynkEventDispatcher(val handler: Handler = Handler(Looper.getMainLooper())) {
+    private val listeners = CopyOnWriteArrayList<UplynkListener>()
+
+    fun dispatchPreplayEvents(response: PreplayResponse) = listeners.forEach { listener ->
+        handler.post { listener.onPreplayResponse(response) }
+    }
```

Comment:
Should we post a callback for each individual listener, or run all listeners in a single post? 🤔 
```suggestion
    fun dispatchPreplayEvents(response: PreplayResponse) = handler.post {
        listeners.forEach { listener -> listener.onPreplayResponse(response) }
    }
```

Reply:
I don't know what's better. I tried to think about different scenarios and to me it's effectively almost the same. Especially because most of the clients would have a single listener.

I checked THEOplayer main code and it loops over listeners in the same single post. I think if it works in main project we better put the same in this.

Reply:
Yes, I suggest we align with THEOplayer's approach. Fewer surprises. 😉